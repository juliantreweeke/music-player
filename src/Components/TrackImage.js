import React, { useEffect, useState } from 'react';
import {Spring} from 'react-spring/renderprops';
import styled from "styled-components";
import { DEFAULT_IMAGE_URL } from '../constants';

const Image = styled.div`
  background: black;
  background-image: url(${props => props.image});
  background-size: cover;
  z-index:-1;
`;

export const TrackImage = ({playing, image}) => {
  let [degree, setDegree] = useState(0);

  const tick = () => {
  setInterval(() => setDegree(degree++ ) , 100)
  }

  const useMountEffect = (func) => useEffect(func, [])

  useMountEffect(tick)

    const imageUrlEdit = image ? image.replace("large", "t500x500") : DEFAULT_IMAGE_URL;

    return (
      <Spring
        to={{
          borderRadius: playing ? "50" : "0",
          scale: playing ? 2 : 1,
          width: playing ? "100%" : "80%",
          containerWidth: playing ? "60%" : "100%",
          margin: playing ? 0 : 0,
          height: playing ? "0%" : "80%",
          rotation: playing ? `${degree}` : "0",
          paddingBottom: playing ? "100" : "80"
        }}
        image={imageUrlEdit}
        playing={playing}
      >
      {props => ( 
          <div
          style={{
            width:props.containerWidth,
            display:'flex',
            justifyContent: 'center'
          }}
          >
          <Image
            image={image}
            style={{
              height:props.height,
              paddingBottom: `${props.paddingBottom}%`,
              width:props.width,
              borderRadius: `${props.borderRadius}%`,
              top: `${props.top}%`,
              margin: `${props.margin}%`,
              left: `${props.left}%`,
              transform: `rotate(${props.rotation}deg)`
            }}
          />
          </div>
      )}
      </Spring>
    );
}


