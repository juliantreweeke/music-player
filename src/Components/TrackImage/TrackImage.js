import React from 'react';
import {Spring} from 'react-spring/renderprops';
import styled from "styled-components";

const Image = styled.div`
  background: black;
  background-image: url(${props => props.image});
  background-size: cover;
  z-index:-1;
  max-width:320;
`;

export const TrackImage = ({degree, playing, image}) => {

    return (
      <Spring
        to={{
          borderRadius: playing ? "50" : "0",
          scale: playing ? 2 : 1,
          width: playing ? "100%" : "80%",
          containerWidth: playing ? "60%" : "100%",
          margin: playing ? 0 : 0,
          height: playing ? "0%" : "80%",
          rotation: playing ? `${degree}` : "0",
          paddingBottom: playing ? "100" : "80"
        }}
        image={image}
        playing={playing}
      >
      {props => ( 
          <div
          style={{
            width:props.containerWidth,
            display:'flex',
            justifyContent: 'center'
          }}
          >
          <Image
            image={image}
            style={{
              height:props.height,
              paddingBottom: `${props.paddingBottom}%`,
              width:props.width,
              borderRadius: `${props.borderRadius}%`,
              top: `${props.top}%`,
              margin: `${props.margin}%`,
              left: `${props.left}%`,
              transform: `rotate(${props.rotation}deg)`
            }}
          />
          </div>
      )}
      </Spring>
    );
}


