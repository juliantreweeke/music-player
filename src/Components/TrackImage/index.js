import React, { useState } from 'react';
import { TrackImage } from './TrackImage';
import { useMountEffect } from '../../utils';
import { DEFAULT_IMAGE_URL } from '../../constants';

export const TrackImageContainer = ({playing, image}) => {
  let [degree, setDegree] = useState(0);
  
  const tick = () => {
  setInterval(() => setDegree(degree++ ) , 100)
  }

  useMountEffect(tick)

  const imageUrlEdit = image ? image.replace("large", "t500x500") : DEFAULT_IMAGE_URL;

   return <TrackImage degree={degree} image={imageUrlEdit} playing={playing}/>
}


