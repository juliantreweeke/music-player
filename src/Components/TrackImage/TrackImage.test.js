import React from 'react';
import { TrackImage } from './TrackImage';
import { cleanup, render} from '@testing-library/react'

import Image from "./testImage.jpg";

describe('TrackImage', () => {
    afterEach(cleanup)     
    
    it("should render", () => {   
         const TrackImageComponent = render(
                <TrackImage playing={false} image={Image} />
                )
          expect(TrackImageComponent).toBeTruthy();  
    });

    it("matches snapshot with playing is false", () => {
        const TrackImageComponent = render(
            <TrackImage playing={false} image={Image} />
            )
            expect(TrackImageComponent).toMatchSnapshot();
    });

    it("matches snapshot with playing is true", () => {
        const TrackImageComponent = render(
            <TrackImage playing={true} image={Image} />
            )
            expect(TrackImageComponent).toMatchSnapshot();
    });

});
