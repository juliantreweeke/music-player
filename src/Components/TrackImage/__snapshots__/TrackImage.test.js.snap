// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackImage matches snapshot with playing is false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; justify-content: center;"
      >
        <div
          class="sc-AxjAm eDDrYY"
          style="border-radius: 0%; width: 70%; margin: 0px; padding-bottom: 70%; transform: rotate(0deg);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; justify-content: center;"
    >
      <div
        class="sc-AxjAm eDDrYY"
        style="border-radius: 0%; width: 70%; margin: 0px; padding-bottom: 70%; transform: rotate(0deg);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TrackImage matches snapshot with playing is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 60%; display: flex; justify-content: center;"
      >
        <div
          class="sc-AxjAm eDDrYY"
          style="border-radius: 50%; width: 100%; margin: 0px; padding-bottom: 100%; transform: rotate(0deg);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 60%; display: flex; justify-content: center;"
    >
      <div
        class="sc-AxjAm eDDrYY"
        style="border-radius: 50%; width: 100%; margin: 0px; padding-bottom: 100%; transform: rotate(0deg);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
